Assets/Scripts/Prefabs/Player.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using UnityEngine;
+
+[RequireComponent(typeof(BoxCollider2D), typeof(Rigidbody2D))]
+public class Player : MonoBehaviour
+{
+    public Action OnAddedPoints;
+    public Action OnJump;
+    public Action OnDied;
+
+    [SerializeField] private Rigidbody2D rb;
+    private float jumpForce;
+
+    [Header("SPRITES")]
+    [SerializeField] private SpriteRenderer head;
+    [SerializeField] private SpriteRenderer body;
+    [SerializeField] private SpriteRenderer wing_L;
+    [SerializeField] private SpriteRenderer wing_R;
+
+    private EventHandler eventHandler;
+
+    public void SetupPlayer(EventHandler eventHandler)
+    {
+        this.eventHandler = eventHandler;
+        SetPlayerData("Player_1");
+        OnJump = Jump;
+        OnDied = Die;
+        eventHandler.AddEventToDict("OnDied", OnDied);
+        eventHandler.AddEventToDict("OnAddedPoints", OnAddedPoints);
+        eventHandler.AddEventToDict("OnJump", OnJump);
+
+    }
+
+    public void InitPlayer()
+    {
+        rb.simulated = true;
+    }
+
+    private void Jump()
+    {
+        rb.AddForce(Vector2.up * jumpForce, ForceMode2D.Impulse);
+    }
+
+    private void Die()
+    {
+        Destroy(gameObject);
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        if (collision.gameObject.tag == "Obstacle")
```

Comment:
You should use CompareTag in these cases, reason: https://discussions.unity.com/t/is-comparetag-better-than-gameobject-tag-performance-wise/33954/2

Also, consider making the string "Obstacle" a constant. It's always preferable to use constants so if you ever need to make a change you'll make in a single place.

Reply:
Thx for the advice. This performance gain seems low for a small project, but every gain is good. 
The constant thing is a technical debt, which I'm planning to change in the future.

Reply:
By the way, if you have any other source to share about unity, I will be grateful.